src/exposurelog/routers/add_message.py

```diff
@@ -128,11 +132,53 @@ async def add_message(
     return Message.from_orm(result)
 
 
-def get_day_obs_from_registries(
+def check_obs_id(obs_id: str, current_day_obs: int) -> None:
+    """Check obs_id.
+
+    Only intended to be called if is_new is true, since otherwise obs_id
+    is checked and the current seq_num retrieved with the butler.
+
+    Parameters
+    ----------
+    obs_id
+        obs_id of the exposure being taken.
+    current_day_obs
+        Current day_obs. Used to check obs_id.
+
+    Returns
+    -------
+    seq_num
+        The exposure sequence number.
+
+    Raises
+    ------
+    fastapi.HTTPException
+        If obs_id has invalid format or if the obs_day field is more
+        than 1 day away from the current_obs_day.
+    """
+    match = OBSID_REGEX.fullmatch(obs_id)
+    if match is None:
+        raise fastapi.HTTPException(
+            status_code=http.HTTPStatus.BAD_REQUEST,
+            detail=f"Invalid {obs_id=}",
+        )
+    obs_id_str = match.groups()[0]
+    if (
+        int(obs_id_str) < current_day_obs - 1
```

Comment:
if you aren't going to call `int` on the line above how about using `(obs_id_int := int(obs_id_str))` here so you aren't calling `int` on the same thing twice in the same `if`?

Reply:
Good catch. Also the variable name is misleading; this is a string version of day_obs, not obs_id. I'll just create day_obs as an integer right away (that is safe because the regex matched).